tests/ttnn/unit_tests/operations/test_moreh_softmax.py

```diff
@@ -326,28 +317,34 @@ def test_softmax_backward_for_dim_nc(shape_dim, compute_kernel_options, device):
 
 @pytest.mark.parametrize(
     "shape_dim_strategy",
-    (
-        ((32, 32), 1, ttnn.operations.moreh.SoftmaxOpParallelizationStrategy.SMALL_W),
-        ((32, 32), 0, ttnn.operations.moreh.SoftmaxOpParallelizationStrategy.SMALL_H),
-        ((32, 32), 1, ttnn.operations.moreh.SoftmaxOpParallelizationStrategy.LARGE_W),
-        ((32, 32), 0, ttnn.operations.moreh.SoftmaxOpParallelizationStrategy.LARGE_H),
-        ((1, 1, 32, 32), 1, ttnn.operations.moreh.SoftmaxOpParallelizationStrategy.LARGE_C),
-        ((1, 1, 32, 32), 0, ttnn.operations.moreh.SoftmaxOpParallelizationStrategy.LARGE_C),
-    ),
+    [
+        [[32, 32], 1, ttnn.operations.moreh.SoftmaxOpParallelizationStrategy.SMALL_W],
+        [[32, 32], 0, ttnn.operations.moreh.SoftmaxOpParallelizationStrategy.SMALL_H],
+        [[32, 32], 1, ttnn.operations.moreh.SoftmaxOpParallelizationStrategy.LARGE_W],
+        [[32, 32], 0, ttnn.operations.moreh.SoftmaxOpParallelizationStrategy.LARGE_H],
+        [[1, 1, 32, 32], 1, ttnn.operations.moreh.SoftmaxOpParallelizationStrategy.LARGE_C],
+        [[1, 1, 32, 32], 0, ttnn.operations.moreh.SoftmaxOpParallelizationStrategy.LARGE_C],
+    ],
 )
-def test_softmax_callback(shape_dim_strategy, device):
-    device.enable_program_cache()
-
+def test_softmax_callback(shape_dim_strategy, device, use_program_cache):
     shape, dim, strategy = shape_dim_strategy
     torch.manual_seed(0)
 
-    x = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16)
+    for i in range(2):
+        x = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16)
 
-    dev_x = ttnn.Tensor(x, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)
+        dev_x = ttnn.Tensor(x, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)
 
-    tt_cpu = torch.softmax(x, dim)
-    for i in range(2):
+        tt_cpu = torch.softmax(x, dim)
+        dev_x = ttnn.Tensor(x, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)
```

Comment:
```Python
    for i in range(2):
        x = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16)

        dev_x = ttnn.Tensor(x, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)

        tt_cpu = torch.softmax(x, dim)
        dev_x = ttnn.Tensor(x, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)
```
It seems that the code for creating dev_x is unnecessarily duplicated twice.

Reply:
Resolved